src/components/TileCatalog/TileCatalog.jsx

```diff
@@ -1,189 +1,176 @@
+  const [tileHeight, setTileHeight] = useState(0);
+
+  const onSize = ({ height }) => {
+    setTileHeight(height);
+  };
+
+  const HeightMonitor = sizeMe({
+    monitorHeight: true,
+  })(({ content }) => <div>{content}</div>);
+
+  const getTile = (rowIdx, colIdx) => {
+    const numTilesPerPage = numRows * numColumns;
+    const tileIndex = rowIdx * numColumns + colIdx + (currentPage - 1) * numTilesPerPage;
+
+    if (tileIndex === 0) {
+      return <HeightMonitor content={tiles[0]} onSize={onSize} />;
+    }
+    return tiles[tileIndex] !== undefined ? (
+      tiles[tileIndex]
+    ) : (
+      <div style={{ height: tileHeight }} />
+    );
+  };
+
+  const renderGrid = () => (
+    <div className="bx--grid">
+      {Array(numRows)
+        .fill(null)
+        .map((i, rowIdx) => (
+          <div className="bx--row">
+            {Array(numColumns)
+              .fill(null)
+              .map((j, colIdx) => (
+                <div className="bx--col">{getTile(rowIdx, colIdx)}</div>
+              ))}
+          </div>
+        ))}
+    </div>
+  );
 
   return (
-    <StyledContainerDiv className={className}>
-      <StyledCatalogHeader>
-        {search && search.placeHolderText ? (
-          <TableToolbarSearch
-            size="sm"
-            value={searchState}
-            labelText={search.placeHolderText}
-            placeHolderText={search.placeHolderText}
-            onChange={handleSearch}
-            id={`${id}-searchbox`}
-          />
-        ) : null}
-      </StyledCatalogHeader>
-      {isLoading ? ( // generate empty tiles for first page
-        <TileGroup
-          tiles={[...Array(pageSize)].map((val, index) => (
-            <StyledEmptyTile key={`emptytile-${index}`}>
-              <SkeletonText />
-            </StyledEmptyTile>
-          ))}
-          totalTiles={totalTiles}
-        />
-      ) : tiles.length > 0 ? (
-        <TileGroup
-          tiles={tiles.map(tile => (
-            <RadioTile
-              className={tile.className}
-              key={tile.id}
-              id={tile.id}
-              value={tile.id}
-              name={id}
-              checked={selectedTileId === tile.id}
-              onChange={onSelection}
-            >
-              {tile.renderContent
-                ? tile.renderContent({ values: tile.values, id: tile.id })
-                : tile.value}
-            </RadioTile>
-          ))}
-        />
-      ) : (
-        <StyledEmptyTile>
-          {error || (
-            <Fragment>
-              <Bee32 />
-              <p>{(search && search.noMatchesFoundText) || 'No matches found'}</p>
-            </Fragment>
-          )}
-        </StyledEmptyTile>
-      )}
-      {!isLoading && tiles.length > 0 && !error && pagination ? (
-        <SimplePagination {...pagination} maxPage={Math.ceil(totalTiles / pageSize)} />
-      ) : null}
-    </StyledContainerDiv>
+    <div>
+      <div>
+        {/* {persistentSearch ? (
```

Comment:
Is this suppose to be removed?

Reply:
These are additions to the component coming soon, I'm fine with leaving them in as comments.